mobile/test/common/extensions/filters/http/network_configuration/network_configuration_filter_test.cc

```diff
@@ -184,6 +186,108 @@ TEST_F(NetworkConfigurationFilterTest, AsyncDnsLookupSuccess) {
   filter_.onLoadDnsCacheComplete(host_info_);
 }
 
+class MockProxyResolver : public Network::ProxyResolver {
+public:
+  MOCK_METHOD(Network::ProxyResolutionResult, resolveProxy,
+              (const std::string& target_url_string, std::vector<Network::ProxySettings>& proxies,
+               std::function<void(std::vector<Network::ProxySettings>& proxies)>
+                   proxy_resolution_did_complete));
+};
+
+class NetworkConfigurationFilterProxyResolverApiTest : public NetworkConfigurationFilterTest {
+public:
+  NetworkConfigurationFilterProxyResolverApiTest() : NetworkConfigurationFilterTest() {}
+
+  void SetUp() override {
+    auto proxy_resolver_api = std::make_unique<Network::ProxyResolverApi>();
+    proxy_resolver_api->resolver = std::make_unique<NiceMock<MockProxyResolver>>();
+    Api::External::registerApi("envoy_proxy_resolver", proxy_resolver_api.release());
```

Comment:
Same as above. Do we want to release the ownership of the `unique_ptr`?

Reply:
the API registry needs to own it for now, unfortunately